src/main/scala/mesosphere/marathon/api/v2/ModelValidation.scala

```diff
@@ -17,6 +17,14 @@ object ModelValidation {
 
   //scalastyle:off null
 
+  /**
+    * This regular expression is used to validate each path segment of an ID.
+    *
+    * If you change this, please also change AppDefinition.json and notify the maintainers of the DCOS CLI.
+    */
+  private[this] val ID_PATH_SEGMENT_PATTERN =
+    "^(([a-z0-9]|[a-z0-9][a-z0-9\\-]*[a-z0-9])\\.)*([a-z0-9]|[a-z0-9][a-z0-9\\-]*[a-z0-9])|(\\.|\\.\\.)$".r
```

Comment:
Can we use (almost, there is a difference I know) the same pattern as the schema?